nservicebus/outbox/index.md

```diff
@@ -79,8 +79,64 @@ In order to gradually convert an entire system from the DTC to the Outbox:
 WARNING: When verifying outbox functionality, it can sometimes be helpful to temporarily [stop the MSDTC Service](https://technet.microsoft.com/en-us/library/cc770732.aspx). This ensures that the Outbox is working as expected, and no other resources are enlisting in distributed transactions.
 
 
+## Message identity and idempotent processing
+
+If handlers are idempotent then outbox might not be required at all. Outbox relies on message identity to guarantee consistency. When using outbox on both the sender and receiver message identifiers are consistent if a handler is invoked more-than-once for a message. 
+If a sender does not use the outbox or when sending a message outside of a handler the sender *must* generate and apply the same message identifier to guarantee consistency at the receiver.
+The same message identity is also important to have deduplication working on the infrastructure level on some transport that support this. [Read more on how to set a message identifier.](/nservicebus/messaging/message-identity.md)
+
+
+## Concurrency
+
+Outbox cannot guarantee that handlers or sagas are invoked exactly once. Outbox does not prevent concurrent attempts to process the same message but ensures that the outcome of the processing is persisted only once for resources that share the outbox transaction.
+
+The following scenarios might cause handlers to be invoked multiple times:
+
+- When a sender sends multiple physical message with the same message identifier. An example is a order form which is submitted multiple times.
+- When a transport performs a failover and operates in a at-least-once delivery model and has not marked a message as processed while this is the case but the acknowledgement was not processed by the tranport yet.
+- When a failure occurs during the outbox dispatch stage. The dispatch stage will to be invoked more-than-once and each time message can message can partially be delivered to the transport infrastructure.
+
+In the above scenarios, multiple physical copies of the same logical message exist in the queue. If an endpoint allows concurrent message processing, or it is scaled out, its possible that the multiple copies of the message can be processed concurrently.
```

Comment:
I think the point about the concurrency is the relevant point. All of the above listed scenarios should be handled correctly in case there is no race condition, so I'm not sure about the value of listing those scenarios and I suggest to focus on the concurrency aspect instead.

Thinking more about these concurrency race conditions I fail to see the larger problem justifying this documentation section. In case of concurrent processing of a duplicate message, one of the messages should fail due to a concurrency exception on the outbox storage. This will rollback the message and keep the "exactly once semantic" of the outbox. Again, the only cases which cause side effects are non-outbox-storage related side-effects, which we cannot prevent in general with the outbox anyway (= this is a know limitation of the outbox).

Reply:
> Again, the only cases which cause side effects are non-outbox-storage related side-effects, which we cannot prevent in general with the outbox anyway (= this is a know limitation of the outbox).